soccer/gameplay/skills/capture.py

```diff
@@ -152,5 +132,37 @@ def role_requirements(self):
         reqs.require_kicking = True
         # try to be near the ball
         if main.ball().valid:
-            reqs.destination_shape = main.ball().pos
+            reqs.cost_func = lambda r: reqs.position_cost_multiplier * find_robot_intercept_point(
+                r).dist_to(r.pos)
+
         return reqs
+
+
+def find_robot_intercept_point(robot):
+    if robot is None:
+        return main.ball().pos
+    approach_vec = approach_vector(robot)
+    # sample every 5 cm in the -approach_vector direction from the ball
+    pos = None
+    for i in range(50):
+        dist = i * 0.05
```

Comment:
I /really/ don't like magic numbers, can you make some global constants for these